### ISSUE TITLE
Release Securedrop 2.5.2

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#6749]
This is a tracking issue for the release of SecureDrop 2.5.2

Tentatively scheduled as follows:

**Pre-release announcement:** 2023-02-16
**Release date:** 2022-02-23

**Release manager:** @zenmonkeykstop  
**Deputy release manager:** @cfm  
**Communications manager:** @nathandyer

_SecureDrop maintainers and testers:_ As you QA 2.5.2, please report back your testing results as comments on this ticket. File GitHub issues for any problems found, tag them "QA: Release".

Test debian packages will be posted on https://apt-test.freedom.press signed with [the test key](https://gist.githubusercontent.com/conorsch/ec4008b111bc3142fca522693f3cce7e/raw/2968621e8ad92db4505a31fcc5776422d7d26729/apt-test%2520apt%2520pubkey).

# [QA Matrix for 2.5.2](https://docs.google.com/spreadsheets/d/18qkpFYs9aHR_q1RDc9KVH96cWmYz_oLmtGh-HK1AkLg/edit#gid=1940373141)
# [Test Plan for 2.5.2](https://github.com/freedomofpress/securedrop/wiki/2.5.2-Test-Plan)

# Prepare release candidate (2.5.2~rc1)
- [ ] Link to latest version of Tails, including release candidates, to test against during QA
- [x] Prepare 2.5.2~rc1 release changelog
- [x] Branch off release/2.5.2 from release/2.5.1
- [x] Prepare 2.5.2~rc1
- [x] Build debs, preserving build log, and put up `2.5.2~rc1` on test apt server
- [x] Commit build log.

After each test, please update the QA matrix and post details for Basic Server Testing, Application Acceptance Testing and release-specific testing below in comments to this ticket.

# Final release
- [ ] Ensure builder in release branch is updated and/or update builder image
- [ ] Push signed tag 
- [ ] Pre-Flight: Test updater logic in Tails (apt-qa tracks the `release` branch in the LFS repo)
- [ ] Build final Debian packages(and preserve build log)
- [ ] Commit package build log to https://github.com/freedomofpress/build-logs
- [ ] Pre-Flight: Test that install and upgrade from 2.5.1 to 2.5.2 works w/ prod repo debs (apt-qa.freedom.press polls the `release` branch in the LFS repo for the debs)
- [ ] Flip apt QA server to prod status (merge to `main` in the LFS repo)
- [ ] Merge Docs branch changes to ``main`` and verify new docs build in securedrop-docs repo
- [ ] Prepare release messaging

# Post release
- [ ] Create GitHub release object 
- [ ] Once release object is created, update versions in `securedrop-docs` and Wagtail
- [ ] Verify new docs show up on https://docs.securedrop.org
- [ ] Publish announcements
- [ ] Merge changelog back to `develop`
- [ ] Update roadmap wiki page: https://github.com/freedomofpress/securedrop/wiki/Development-Roadmap

**Comment by cfm** [comment]
@zenmonkeykstop, I've drafted the QA matrix and test plan (now linked above).  Based on #6753, there don't seem to be any release-specific test steps, since everything is either a dependency update or checked by testinfra.  Let me know if you disagree.

**Comment by cfm** [comment]
### Environment

- Install target: NUC11s
- Tails version: 5.7
- Test Scenario: upgrade
- SSH over Tor: yes
- Release candidate: `2.5.2~rc1`
- General notes:

### Basic Server Testing
- [ ] After installing the testinfra dependencies, all tests in `./securedrop-admin verify` are passing:
  - Install dependencies on Admin Workstation with `cd ~/Persistent/securedrop && ./securedrop-admin setup -t`
  - Run tests with `./securedrop-admin verify` (this will take a while)
  - Remove test dependencies: `rm -rf admin/.venv3/ && ./securedrop-admin setup`

**No,** with these failures:
- `test_app_pip_deps`: 9a741b54d0fe0401942df08ac24f604046b4c3cd cherry-picked from #6577 needs #6634
- `test_apache_default_docroot_is_absent`: resolved after `securedrop-admin install` per <https://github.com/freedomofpress/securedrop/issues/6749#issuecomment-1441162687>
```sh-session
(securedrop-app-code) sdadmin@app:~$ ls -al /var/www
total 24
drwxr-xr-x  4 root root 4096 Feb 22 18:39 .
drwxr-xr-x 14 root root 4096 Mar 22  2022 ..
drwxr-xr-x  2 root root 4096 Jan 23 18:36 html
-rw-r--r--  1 root root  262 Feb 21 21:10 journalist.wsgi
drwxr-xr-x 16 root root 4096 Feb 22 18:39 securedrop
-rw-r--r--  1 root root  261 Feb 21 21:10 source.wsgi
```
- kernel-version checks (expected)

- [x] QA Matrix checks pass

#### Command Line User Generation

- [x] Can successfully add admin user and login

#### (Optional) Administration

- [-] I have backed up and successfully restored the app server following [the backup documentation](https://docs.securedrop.org/en/latest/backup_and_restore.html)
- [-] If doing upgrade testing, make a backup on 2.5.1 and restore this backup on this release candidate
- [x] "Send Test OSSEC Alert" button in the journalist triggers an OSSEC alert and an email is sent
- [x] Can successfully add journalist account with HOTP authentication

### (Optional) Application Acceptance Testing

#### Source Interface

##### Landing page base cases
- [x] JS warning bar does not appear when using Security Slider high
- [x] JS warning bar does appear when using Security Slider Low

##### First submission base cases
- [x] On generate page, refreshing page produces a new 7-word codename
- [x] On submit page, empty submissions produce flashed message
- [x] On submit page, short message submitted successfully
- [x] On submit page, file greater than 500 MB produces "The connection was reset" in Tor Browser _quickly_ before the entire file is uploaded
- [x] On submit page, file less than 500 MB submitted successfully

##### Returning source base cases

- [x] Nonexistent codename cannot log in
- [x] Empty codename cannot log in
- [x] Legitimate codename can log in
- [x] Returning user can view journalist replies - need to log into journalist interface to test

#### Journalist Interface

##### Login base cases
- [x] Can log in with 2FA tokens
- [x] incorrect password cannot log in
- [x] invalid 2fa token cannot log in
- [x] 2fa immediate reuse cannot log in
- [x] Journalist account with HOTP can log in

##### Index base cases

- [x] Filter by codename works
- [x] Starring and unstarring works
- [x] Click select all selects all submissions
- [x] Selecting all and clicking "Download" works

##### Individual source page

- [x] You can submit a reply and a flashed message and new row appears
- [x] You cannot submit an empty reply
- [x] Clicking "Delete Source Account" and the source and docs are deleted
- [x] You can click on a document and successfully decrypt using application private key

### Basic Tails Testing

After updating to this release candidate and running `securedrop-admin tailsconfig`

- [x] The Updater GUI appears on boot

### 2.5.2 release-specific changes

*None*

**Comment by cfm** [comment]
<https://github.com/freedomofpress/securedrop/issues/6749#issuecomment-1440804168>:

> * `test_apache_default_docroot_is_absent`: resolved after `securedrop-admin install`; ticket TK
> 
> 
> ```
> (securedrop-app-code) sdadmin@app:~$ ls -al /var/www
> total 24
> drwxr-xr-x  4 root root 4096 Feb 22 18:39 .
> drwxr-xr-x 14 root root 4096 Mar 22  2022 ..
> drwxr-xr-x  2 root root 4096 Jan 23 18:36 html
> -rw-r--r--  1 root root  262 Feb 21 21:10 journalist.wsgi
> drwxr-xr-x 16 root root 4096 Feb 22 18:39 securedrop
> -rw-r--r--  1 root root  261 Feb 21 21:10 source.wsgi
> ```

My hypothesis is that `/var/www/html` was created on February 2 when `apache2` was upgraded to version `2.4.41-4ubuntu3.13`.  I.e., `/var/www/html` is sometimes recreated by `apache2`'s `postinst` script; but it's not removed by `securedrop-app-code`'s `postinst` script, only by `securedrop-admin install`'s Ansible run.  @zenmonkeykstop, is this worth opening a ticket for?

**Comment by nathandyer** [comment]
### Environment

- Install target: NUC12 (App) and NUC11 (Mon)
- Tails version: 5.8
- Test Scenario: Clean Install
- SSH over Tor: yes
- Release candidate: `2.5.2~rc1`
- General notes:

### Basic Server Testing
- [ ] After installing the testinfra dependencies, all tests in `./securedrop-admin verify` are passing:
    **No,** had the same `test_app_pip_deps` error that @cfm encountered. Did not encounter the same `test_apache_default_docroot_is_absent` error, however I only did this verification run _after_ the `install` Ansible run, so that's probably why.

- [x] QA Matrix checks pass

#### Command Line User Generation

- [x] Can successfully add admin user and login

#### (Optional) Administration

<details><summary><em>Skipped</em></summary>

- [ ] I have backed up and successfully restored the app server following [the backup documentation](https://docs.securedrop.org/en/latest/backup_and_restore.html)
- [ ] If doing upgrade testing, make a backup on 2.5.1 and restore this backup on this release candidate
- [ ] "Send Test OSSEC Alert" button in the journalist triggers an OSSEC alert and an email is sent
- [ ] Can successfully add journalist account with HOTP authentication

</details>

### (Optional) Application Acceptance Testing

<details><summary><em>Skipped</em></summary>

#### Source Interface

##### Landing page base cases
- [ ] JS warning bar does not appear when using Security Slider high
- [ ] JS warning bar does appear when using Security Slider Low

##### First submission base cases
- [ ] On generate page, refreshing page produces a new 7-word codename
- [ ] On submit page, empty submissions produce flashed message
- [ ] On submit page, short message submitted successfully
- [ ] On submit page, file greater than 500 MB produces "The connection was reset" in Tor Browser _quickly_ before the entire file is uploaded
- [ ] On submit page, file less than 500 MB submitted successfully

##### Returning source base cases

- [ ] Nonexistent codename cannot log in
- [ ] Empty codename cannot log in
- [ ] Legitimate codename can log in
- [ ] Returning user can view journalist replies - need to log into journalist interface to test

#### Journalist Interface

##### Login base cases
- [ ] Can log in with 2FA tokens
- [ ] incorrect password cannot log in
- [ ] invalid 2fa token cannot log in
- [ ] 2fa immediate reuse cannot log in
- [ ] Journalist account with HOTP can log in

##### Index base cases

- [ ] Filter by codename works
- [ ] Starring and unstarring works
- [ ] Click select all selects all submissions
- [ ] Selecting all and clicking "Download" works

##### Individual source page

- [ ] You can submit a reply and a flashed message and new row appears
- [ ] You cannot submit an empty reply
- [ ] Clicking "Delete Source Account" and the source and docs are deleted
- [ ] You can click on a document and successfully decrypt using application private key

</details>

### Basic Tails Testing

After updating to this release candidate and running `securedrop-admin tailsconfig`

- [x] The Updater GUI appears on boot

### 2.5.2 release-specific changes

*None*

**Comment by zenmonkeykstop** [comment]
> My hypothesis is that `/var/www/html` was created on February 2 when `apache2` was upgraded to version `2.4.41-4ubuntu3.13`. I.e., `/var/www/html` is sometimes recreated by `apache2`'s `postinst` script; but it's not removed by `securedrop-app-code`'s `postinst` script, only by `securedrop-admin install`'s Ansible run. @zenmonkeykstop, is this worth opening a ticket for?

There's an existing ticket elsewhere for this - we could probably make it an issue in the main repo tho after checking with @l3th3 or @lsd-cat . Your analysis is right on the money. Also, it's an existing issue, not a regression, so I would argue not a release blocker.

**Comment by zenmonkeykstop** [comment]
## 2.5.2 QA Checklist

### Environment

- Install target: PRod VMs 
- Tails version: 5.9
- Test Scenario: fresh install 
- SSH over Tor: no
- Release candidate: rc1
- General notes:

### Basic Server Testing
- [x] After installing the testinfra dependencies, all tests in `./securedrop-admin verify` are passing: **one fail due to new Flask version**
  - Install dependencies on Admin Workstation with `cd ~/Persistent/securedrop && ./securedrop-admin setup -t`
  - Run tests with `./securedrop-admin verify` (this will take a while)
  - Remove test dependencies: `rm -rf admin/.venv3/ && ./securedrop-admin setup`
- [x] QA Matrix checks pass

#### Command Line User Generation

- [x] Can successfully add admin user and login

#### (Optional) Administration

- [x] I have backed up and successfully restored the app server following [the backup documentation](https://docs.securedrop.org/en/latest/backup_and_restore.html)
- [ ] If doing upgrade testing, make a backup on 2.5.1 and restore this backup on this release candidate **N/A**
- [x] "Send Test OSSEC Alert" button in the journalist triggers an OSSEC alert and an email is sent
- [ ] Can successfully add journalist account with HOTP authentication**NOT TESTED**

### (Optional) Application Acceptance Testing

#### Source Interface

##### Landing page base cases
- [x] JS warning bar does not appear when using Security Slider high
- [x] JS warning bar does appear when using Security Slider Low

##### First submission base cases
- [x] On generate page, refreshing page produces a new 7-word codename
- [x] On submit page, empty submissions produce flashed message
- [x] On submit page, short message submitted successfully
- [x] On submit page, file greater than 500 MB produces "The connection was reset" in Tor Browser _quickly_ before the entire file is uploaded
- [x] On submit page, file less than 500 MB submitted successfully

##### Returning source base cases

- [x] Nonexistent codename cannot log in
- [x] Empty codename cannot log in
- [x] Legitimate codename can log in
- [x] Returning user can view journalist replies - need to log into journalist interface to test

#### Journalist Interface

##### Login base cases
- [x] Can log in with 2FA tokens
- [x] incorrect password cannot log in
- [x] invalid 2fa token cannot log in
- [x] 2fa immediate reuse cannot log in
- [ ] Journalist account with HOTP can log in

##### Index base cases

- [x] Filter by codename works
- [x] Starring and unstarring works
- [x] Click select all selects all submissions
- [x] Selecting all and clicking "Download" works

##### Individual source page

- [x] You can submit a reply and a flashed message and new row appears
- [x] You cannot submit an empty reply
- [x] Clicking "Delete Source Account" and the source and docs are deleted
- [x] You can click on a document and successfully decrypt using application private key

### Basic Tails Testing

After updating to this release candidate and running `securedrop-admin tailsconfig`

- [x] The Updater GUI appears on boot

### 2.5.2 release-specific changes

*None*

### Preflight testing

#### Basic testing

- [x] Install or upgrade occurs without error (from `apt-qa.freedom.press` per [preflight procedure](https://github.com/freedomofpress/securedrop/wiki/QA-Procedures#pre-release-prep))
- [x] Source interface is available and version string indicates it is 2.5.2
- [x] A message can be successfully submitted

#### Tails

- [ ] The updater GUI appears on boot
- [ ] The update successfully occurs to 2.5.2
- [ ] After reboot, updater GUI no longer appears

**Comment by cfm** [comment]
<https://github.com/freedomofpress/securedrop/issues/6749#issuecomment-1440804168> updated with acceptance tests, which pass.

**Comment by cfm** [comment]
### Environment

- Install target: NUC11s
- Tails version: 5.10
- Test Scenario: upgrade
- SSH over Tor: yes
- Release candidate: `2.5.2`
- General notes:

### Preflight testing

#### Basic testing

- [x] Install or upgrade occurs without error (from `apt-qa.freedom.press` per [preflight procedure](https://github.com/freedomofpress/securedrop/wiki/QA-Procedures#pre-release-prep))
- [x] Source interface is available and version string indicates it is 2.5.2
- [x] A message can be successfully submitted

#### Tails

- [x] The updater GUI appears on boot
- [x] The update successfully occurs to 2.5.2
- [x] After reboot, updater GUI no longer appears

**Comment by zenmonkeykstop** [comment]
Preflight checks look good for fresh install
